src/topology/metric_space/basic.lean

```diff
@@ -1562,6 +1577,19 @@ theorem closed_ball_prod_same (x : α) (y : β) (r : ℝ) :
   closed_ball x r ×ˢ closed_ball y r = closed_ball (x, y) r :=
 ext $ λ z, by simp [prod.dist_eq]
 
+theorem sphere_prod (x : α) (y : β) (r : ℝ) :
+  sphere (x, y) r = sphere x r ×ˢ closed_ball y r ∪ closed_ball x r ×ˢ sphere y r :=
```

Comment:
To match the pi result, you should take in `x : \alpha \times \beta`

Reply:
I've written it this way to match the prod lemmas immediately above

Reply:
That's different though. The one above is meant to rewrite from a product of balls to a ball in the product. Yours is meant to unfold the definition of a sphere in the product (and the corresponding lemma doesn't exist for balls). So I maintain you should take in a `prod`.